libfreerdp/core/freerdp.c

```diff
@@ -238,11 +238,17 @@ BOOL freerdp_connect(freerdp* instance)
 
 BOOL freerdp_abort_connect(freerdp* instance)
 {
-	if (!instance || !instance->context)
-		return FALSE;
+	WINPR_ASSERT(instance);
```

Comment:
@mfleisz don´t agree. The function should return if the abort did work (e.g. the event was set)
Misusing the API should not be tolerated.

Reply:
@akallabeth 
Since the function returns a `BOOL` I assume it will return `TRUE` if setting the event succeeded or `FALSE` if something (i.e. an invalid input parameter) went wrong

If I pass a `NULL` pointer to this function in an optimized release build (which removes all asserts) it will crash at line 243 (same for all the other mentioned functions)

In my opinion a public API should be safe (not crash the library code) against invalid usage.